nimiSlides.nim

```diff
@@ -56,66 +74,64 @@ type
     highlightCurrentRed = "highlight-current-red"
     highlightCurrentGreen = "highlight-current-green"
     highlightCurrentBlue = "highlight-current-blue"
-  Slide* = ref object
-    pos*: tuple[start: int, finish: int]
-    notes*: seq[string]
-  SlidesCtx* = ref object
-    sections*: seq[seq[Slide]]
+
   SlidesTheme* = enum
     Black, Beige, Blood, League, Moon, Night, Serif, Simple, Sky, Solarized, White
 
-################################
-# This is modified code from nimib.
-# We need to render code blocks using the context to populate the {{ endFragment-i }} fields.
-################################
-
-proc renderHtmlTextOutput*(output: string, doc: NbDoc): string =
-  renderMarkdown(output.strip).render(doc.context)
-
-proc renderHtmlBlock*(blk: NbBlock, doc: NbDoc): string =
-  case blk.kind
-  of nbkText:
-    result = blk.output.renderHtmlTextOutput(doc)
-  of nbkCode:
-    result = blk.code.renderHtmlCodeBody
-    if blk.output != "":
-      result.add blk.output.renderHtmlCodeOutput
-  of nbkImage:
-    let
-      image_url = blk.code
-      caption = blk.output
-    result = fmt"""
-<figure>
-<img src="{image_url}" alt="{caption}">
-<figcaption>{caption}</figcaption>
-</figure>
-""" & "\n"
-###########################
-###########################
-
 template initReveal*() =
   ## Call this after nbInit
-  var slidesCtx {.inject.} = SlidesCtx(sections: @[@[Slide(pos: (start: 0, finish: -1))]])
-  nb.context["currentFragment"] = 0
-  nb.context["currentEndFragment"] = 0
-
-  template slideRight() =
-    ## Add a slide to the right of the current one
-    slidesCtx.sections[^1][^1].pos.finish = nb.blocks.len - 1
-    slidesCtx.sections.add @[Slide(pos: (start: nb.blocks.len, finish: -1))]
-
-  template slideDown() =
-    ## Add a slide below the current one
-    slidesCtx.sections[^1][^1].pos.finish = nb.blocks.len - 1
-    slidesCtx.sections[^1].add (Slide(pos: (start: nb.blocks.len, finish: -1)))
-
-  template slideRight(body: untyped) =
-    slideRight()
-    body
-  
-  template slideDown(body: untyped) =
-    slideDown()
-    body
+  var currentFragment: int
+  when (NimMajor, NimMinor, NimPatch) > (1, 6, 0):
+    {.experimental: "flexibleOptionalParams".}
+
+  template rawBlock(body: untyped) =
```

Comment:
In nimib with last release I started moving all templates outside of `nbInit`. In this way you will not see an unused warning everytime you do not use a template imported through nbInit. I think it should be safe to do it this way also in nimislides but I am not that big of an expert on templates. You would have to export them of course (you might want to skip the eporting of internal fragment blocks (but it would not hurt that much)

Reply:
Yes, was there I got the inspiration from. it should work fine to move them out. If I manage to get away with the `experimental`, the init call should be totally redundant and removed.